ilastik/applets/dataSelection/dataSelectionSerializer.py

```diff
@@ -260,6 +257,12 @@ def _readDatasetInfo(self, infoGroup, projectFilePath, headless):
                 info_class = PreloadedArrayDatasetInfo
 
         dirty = False
+
+        if info_class == UrlDatasetInfo:
+            if not _confirmLoadLegacyProject():
+                raise RuntimeError("Loading aborted by user.")
```

Comment:
Just a small comment, but from a user perspective I find it a bit odd to first get a popup that implies a choice (load project or not) and if I go to abort, I'll get hit with a `RuntimeError` + messagebox:


### 1
<img width="532" alt="image" src="https://github.com/ilastik/ilastik/assets/24434157/476f578f-7dd1-42c0-a4b6-8b256e6a059d">

### 2
<img width="415" alt="image" src="https://github.com/ilastik/ilastik/assets/24434157/63912eac-4a76-40cc-8d47-23d011066034">

Maybe some other error class could be raised here and to prevent the popup in `ilastikShell.py` https://github.com/ilastik/ilastik/blob/9058943988a8257b932c1d9f907ce9f3f9c29d71/ilastik/shell/gui/ilastikShell.py#L1671